submissions/keiran/scripts/articleView.js

```diff
@@ -0,0 +1,108 @@
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $('article[data-category="' + $(this).val() + '"]').fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = function() {
+  $('.main-nav').on('click', '.tab', function(e) {
+    $('.tab-content').hide();
+    $('#' + $(this).data('content')).fadeIn();
+  });
+
+  $('.main-nav .tab:first').click(); // Let's now trigger a click on the first .tab element, to set up the page.
+};
+
+articleView.setTeasers = function() {
+  $('.article-body *:nth-of-type(n+2)').hide(); // Hide elements beyond the first 2 in any artcile body.
+
+  $('#articles').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    $(this).parent().find('*').fadeIn();
+    $(this).hide();
+  });
+};
+
+articleView.initNewArticlePage = function() {
+  // TODO: Ensure the main .tab-content area is revealed. We might add more tabs later.
+  $('.tab-content').show();
+  // TODO: The new articles we create will be copy/pasted into our source data file.
+  // Set up this "export" functionality. We can hide it for now, and show it once we
+  // have data to export. Also, let's add a focus event to help us select the JSON.
+  $('#export-field').hide();
+  $('#article-json').on('focus', function() {
+    $(this).select();
+  });
+  // TODO: Add an event handler to update the preview and the export field if any inputs change.
+  $('#new-form').on('change', 'input', 'textarea', articleView.create);
+};
+
+articleView.create = function() {
+  // TODO: Set up a var to hold the new article we are creating.
+  var formArticle = new Article({
+    title: $('#article-title').val(),
+    body: $('#article-body').val(),
+    author: $('#article-author').val(),
+    authorURL: $('#article-author-url').val(),
+    category: $('#article-category').val(),
+    publishedOn: $('#article-published:checked').length ? new Date() : null
+  });
+  // Clear out the #articles element, so we can put in the updated preview
+  $('#articles').empty();
+  // TODO: Instantiate an article based on what's in the form fields:
+  // DONE ABOVE
+  // TODO: Since formArticle is an instance of the Article constructor,
+  // we can use the .toHtml method on it to append this new article
+  // into the DOM:
+  $('#articles').append(formArticle.toHtml('#article-template'));
+  // TODO: Activate the highlighting of any code blocks:
+  $('pre code').each(function(i, block) {
+    hljs.highlightBlock(block);
+  });
+  // TODO: Export the new article as JSON, so it's ready to copy/paste into blogArticles.js:
+  $('#export-field').show();
+  $('#article-json').val(JSON.stringify(formArticle + ','));
```

Comment:
your #article-json is showing "[Object, Object]", does the comma need to be stringified?